app/src/main/java/com/android/sample/ui/swipePage/SwipePage.kt

```diff
@@ -406,44 +413,57 @@ fun RecipeDisplay(
                 ShawRecipeButton(navigationActions)
               }
 
-              LaunchedEffect(offsetXCard1.value) {
+              LaunchedEffect(offsetXCard1.value, isClicking) {
                 animateCard(
                     offsetX = offsetXCard1,
                     screenWidth = screenWidth,
                     isClicking = isClicking,
-                    isSwiping = isSwiping,
                     retrieveNextRecipe = retrieveNextRecipe,
                     recipesViewModel = recipesViewModel,
-                    startSwiping = { isSwiping = true },
-                    endSwiping = { isSwiping = false },
                     updateDisplayCard = { displayCard1 = false },
                     displayIcons = { animationTarget ->
                       displayLike = animationTarget == END_ANIMATION
                       displayDisLike = animationTarget == -END_ANIMATION
                     },
-                    blockRetrieveNextRecipe = { retrieveNextRecipe = false })
+                    blockRetrieveNextRecipe = { retrieveNextRecipe = false },
+                    startSwipe = { isSwiping = true },
+                    endSwipe = { isSwiping = false })
               }
-              LaunchedEffect(offsetXCard2.value) {
+              LaunchedEffect(offsetXCard2.value, isClicking) {
                 animateCard(
                     offsetX = offsetXCard2,
                     screenWidth = screenWidth,
                     isClicking = isClicking,
-                    isSwiping = isSwiping,
                     retrieveNextRecipe = retrieveNextRecipe,
                     recipesViewModel = recipesViewModel,
-                    startSwiping = { isSwiping = true },
-                    endSwiping = { isSwiping = false },
                     updateDisplayCard = { displayCard1 = true },
                     displayIcons = { animationTarget ->
                       displayLike = animationTarget == END_ANIMATION
                       displayDisLike = animationTarget == -END_ANIMATION
                     },
-                    blockRetrieveNextRecipe = { retrieveNextRecipe = true })
+                    blockRetrieveNextRecipe = { retrieveNextRecipe = true },
+                    startSwipe = { isSwiping = true },
+                    endSwipe = { isSwiping = false })
               }
             }
       }
 }
 
+/**
+ * limit the swipe gesture if the user is spamming the swipe
+ *
+ * @param offsetX - Offset for the swipe animation
+ * @param dragAmount - Drag amount
+ * @param swipeThreshold - Swipe Threshold
+ */
+private fun getTarget(
+    offsetX: Animatable<Float, AnimationVector1D>,
+    dragAmount: Float,
+    swipeThreshold: Float
+) =
+    if (offsetX.value > 0) max(offsetX.value + dragAmount, swipeThreshold)
+    else min(offsetX.value + dragAmount, -swipeThreshold)
```

Comment:
I am not really a big fan of this syntax. I have a feeling that it could bring more bugs in the future if we add some operations to the if condition for example. What do you think? Would it be better to have the curly brackets with the returns or not?

Reply:
I agree; I'll update it by adding brackets.

Reply:
Ok.